ci/windows_dev_env/windows_deps_headless_installer.py

```diff
@@ -0,0 +1,429 @@
+    logging.info("Installing Clang")
+    with tempfile.TemporaryDirectory() as tmpdir:
+        clang_file_path = download(DEPS['clang'], tmpdir)
+        run_command(clang_file_path + " /S /D=C:\\Program Files\\LLVM")
+    logging.info("Clang install complete")
+
+
+def install_openblas():
+    logging.info("Installing OpenBLAS")
+    local_file = download(DEPS['openblas'])
+    with zipfile.ZipFile(local_file, 'r') as zip:
+        zip.extractall("C:\\Program Files")
+    run_command("PowerShell Set-ItemProperty -path 'hklm:\\system\\currentcontrolset\\control\\session manager\\environment' -Name OpenBLAS_HOME -Value 'C:\\Program Files\\OpenBLAS-windows-v0_2_19'")
+    logging.info("Openblas Install complete")
+
+
+def install_mkl():
+    logging.info("Installing MKL 2019.3.203...")
+    file_path = download("http://registrationcenter-download.intel.com/akdlm/irc_nas/tec/15247/w_mkl_2019.3.203.exe")
+    run_command("{} --silent --remove-extracted-files yes --a install -output=C:\mkl-install-log.txt -eula=accept".format(file_path))
+    logging.info("MKL Install complete")
+
+
+def install_opencv():
+    logging.info("Installing OpenCV")
+    with tempfile.TemporaryDirectory() as tmpdir:
+        local_file = download(DEPS['opencv'])
+        with zipfile.ZipFile(local_file, 'r') as zip:
+            zip.extractall(tmpdir)
+        copy(f'{tmpdir}\\opencv\\build', r'c:\Program Files\opencv')
+
+    run_command("PowerShell Set-ItemProperty -path 'hklm:\\system\\currentcontrolset\\control\\session manager\\environment' -Name OpenCV_DIR -Value 'C:\\Program Files\\opencv'")
+    logging.info("OpenCV install complete")
+
+
+def install_cudnn():
+    # cuDNN
+    logging.info("Installing cuDNN")
+    with tempfile.TemporaryDirectory() as tmpdir:
+        local_file = download(DEPS['cudnn'])
+        with zipfile.ZipFile(local_file, 'r') as zip:
+            zip.extractall(tmpdir)
+        copy(tmpdir+"\\cuda\\bin\\cudnn64_7.dll", "C:\\Program Files\\NVIDIA GPU Computing Toolkit\\CUDA\\v9.2\\bin")
+        copy(tmpdir+"\\cuda\\include\\cudnn.h", "C:\\Program Files\\NVIDIA GPU Computing Toolkit\\CUDA\\v9.2\\include")
+        copy(tmpdir+"\\cuda\\lib\\x64\\cudnn.lib", "C:\\Program Files\\NVIDIA GPU Computing Toolkit\\CUDA\\v9.2\\lib\\x64")
+    logging.info("cuDNN install complete")
+
+
+def install_gpu_driver(force=False):
+    if has_gpu() or force:
+        logging.info("GPU detected")
+        install_nvdriver()
+
+
+def install_nvdriver():
+    logging.info("Installing Nvidia Display Drivers...")
+    with tempfile.TemporaryDirectory(prefix='nvidia drivers') as tmpdir:
+        local_file = download(DEPS['nvdriver'])
+        with zipfile.ZipFile(local_file, 'r') as zip:
+            zip.extractall(tmpdir)
+        with remember_cwd():
+            os.chdir(tmpdir)
+            check_call(".\setup.exe -noreboot -clean -noeula -nofinish -passive")
+    logging.info("NVidia install complete")
+
+
+def install_cuda():
+    # CUDA 9.2 and patches
```

Comment:
Enforcing Cuda 9.2 may not be very future proof for this script.

Reply:
CUDA is updated to 10.2 now.